### PR TITLE
frontends.torch.Tensor: add matmul instance method

### LINKED ISSUE #6557: matmul

### PR DESCRIPTION
**Pull request by hmahmood24**
Close #6557

### CODE REVIEW

**Comment by hmahmood24** [comment]
Hi,
Regarding the failing test, I added `@reproduce_failure('6.55.0', b'AXicY2BgYGRgBCIGOGAaKswGCRCL8SIqczC4jHIm1EMXcCtQACvA7eMGZXwKALhvB3k=')` to my test and bumped the hypothesis version to **6.55.0** as mentioned in the decorator, but was unable to reproduce the error getting:

> **hypothesis.errors.DidNotReproduce: Expected the test to raise an error, but it completed successfully**

Just thought you should know.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2587]: where are `as_variable` and `native_array` defined?

**Review by simonetgordon — CHANGES_REQUESTED** [review]

**Comment by simonetgordon** [comment]
> 

Hi there, thanks for the contribution! And thanks for letting me know about this. Please see me comments regarding the test you've written.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2587]: The `@handle_cmd_line_args` decorator takes care of these strategies at runtime so there's no need to explicitly define them in the test. I have never had to define them previously whenever I have contributed to the codebase. Additionally, you can read point no. 3 [here](https://github.com/unifyai/ivy/pull/5150#issuecomment-1264541807). One of the maintainers of this repo actually advised me against defining strategies for `as_variable` and `native_array` and let the `@handle_cmd_line_args` take care of that for us. I don't think anyone else has defined them explicitly either.

Comment (simonetgordon) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2587]: Hi @hmahmood24, indeed you're correct, my apologies, I haven't been working on the frontends so I'm a little behind with some of the strategies updates.

As for the the assertion error, perhaps try adjusting the `rtol` and see if that makes a difference, although I'm concerned about the tolerance being high already. This issue is surprising, especially given the ground truth reference is torch also.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2583]: `.example()` should not be called in the data generation. Perhaps you kept this in whilst testing?

Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: Hi @simonetgordon, yeah, my bad. I forgot to remove `.example()` from the test itself before committing the code. I have removed that, as well as `rtol` itself and tried fixing the tests. They run perfectly fine on my machine against **all** backends as of the latest commit. Looking forward to your feedback now.

Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: Hold up.

Someone refactored the entire testing criteria a few hours ago and made major changes to the frontend tests. I took the latest pull to merge any conflicts with my latest commits but when I try to run my tests, hypothesis fails throwing all sorts of weird errors. It isn't even able to collect the tests, running them is a different story altogether. I think I will wait out till these new changes are made stable enough and the errors are fixed on the `main` branch before running tests specific to just my function. 

@simonetgordon can you please have someone look into the `ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py` file so they can fix these errors as a consequence of the recent refactoring? Thanks!

Comment (simonetgordon) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: Hi @hmahmood24, yes I am aware of the recent frontend decorator changes, apologies for the inconvenience! I will let you know when they've been fixed.

Since you said the tests worked fine for you locally, hopefully we only need to wait for the decorator issues to be resolved to get this merged. If it takes too long, I will merge this PR within a reasonable amount of time.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (simonetgordon) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2579]: we now have 4 new decorators instead of `handle_cmd_line_args` :

1. `handle_function`: used for Ivy functional API
2. `handle_method`: used for Ivy methods
3. `handle_frontend_function`: used for frontend functions
4. `handle_frontend_method`: used for frontend methods, methods tests are now skipped temporarily.

The last decorator is the one you'll need to replace `handle_cmd_line_args` with. Unfortunately, this is the only one currently not functioning correctly, but it will be rectified soon!

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (simonetgordon) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2732]: @hmahmood24 perhaps you didn't see this update?

Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2732]: @simonetgordon,

Apologies I wasn't able to reply earlier. I have seen the updates and I have also refactored my tests in coherence with these updates but to no avail. The `test_tensor.py` file has still so many bugs that prevents the tests from getting collected. At this point, I'm not sure what's the best course of action for me apart from just waiting this out.

Have you asked anyone from the team to look into these bugs by now?

Comment (simonetgordon) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2732]: @hmahmood24 Hi there, yes someone is trying to tackle these but there should have already been some fixes pushed. 

Also, I don't see the new decorator here in the PR version. Perhaps you forgot to push?

**Review by simonetgordon — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py
Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: @simonetgordon Hey, yes I didn't push my changes since the tests weren't collecting locally. I have made the changes and pushed them to remain coherent with all the other tests now.

Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py]: Hi @simonetgordon,

Any updates since it’s been almost 2 weeks since this PR got opened? Anything from my end to add here?

**Review by simonetgordon — APPROVED** [review]

**Pull request merged**
